Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/MDNormSCDPreprocessIncoherent.py

```diff
@@ -107,14 +115,23 @@ def PyExec(self):
              FilterByTofMin=self.getProperty("FilterByTofMin").value,
              FilterByTofMax=self.getProperty("FilterByTofMax").value)
 
+        if _norm_current:
+            NormaliseByCurrent(InputWorkspace='__van',
+                               OutputWorkspace='__van')
+
         if _background:
             Load(Filename=self.getProperty("Background").value,
                  OutputWorkspace='__bkg',
                  FilterByTofMin=self.getProperty("FilterByTofMin").value,
                  FilterByTofMax=self.getProperty("FilterByTofMax").value)
-            pc_van = mtd['__van'].run().getProtonCharge()
-            pc_bkg = mtd['__bkg'].run().getProtonCharge()
-            mtd['__bkg'] *= self.getProperty('BackgroundScale').value*pc_van/pc_bkg
+            if _norm_current:
+                NormaliseByCurrent(InputWorkspace='__bkg',
+                                   OutputWorkspace='__bkg')
+            else:
+                pc_van = mtd['__van'].run().getProtonCharge()
+                pc_bkg = mtd['__bkg'].run().getProtonCharge()
+                mtd['__bkg'] *= pc_van/pc_bkg
+            mtd['__bkg'] *= self.getProperty('BackgroundScale')
```

Comment:
`mtd['__bkg'] *= self.getProperty('BackgroundScale')` should be `mtd['__bkg'] *= self.getProperty('BackgroundScale').value`